README.md

```diff
@@ -94,7 +94,7 @@ On a unix system, the build directory should now look something like this:
 Next, you can run the tests to check the build.
 
 ```bash
-$ cmake --build "build" --config Release --target test
+$ cmake -E chdir "build" ctest ../
```

Comment:
I don't have experience with CMake 3.5 (which is the minimum version required by `CMakeLists.txt`), however, based on [ctest(1)](https://cmake.org/cmake/help/v3.5/manual/ctest.1.html) for this version, I think this is mostly OK.

However, I would put it this way:
```bash
cmake -E chdir "build" ctest --build-config Release
```
where:
- `--build-config <cfg>` - you need this for compatibility with build systems supporting multi-configuration (like for example Visual Studio).

You don't need this `../`. I guess it is ignored since I don't see any error messages when passing it.

There is also `--parallel <jobs>`. However, with a single testing binary, it doesn't add anything until `include(GoogleTest)` is used to promote single test cases to tests from CTest point of view.

---

I just checked the full procedure (steps in `README.md` + the command here) on Windows, with Visual Studio 16 2019 and CMake 3.15.1 and it does work. Tomorrow I could test it on Linux.

---

However, I noticed that from time to time I'm getting
```plain
      Start 57: complexity_benchmark
57/62 Test #57: complexity_benchmark .......................Exit code 0xc0000409
***Exception:   2.38 sec
```
It seems this test is toggling - is it a known issue?

Reply:
One more thing, after doing
```bash
cmake -DCMAKE_BUILD_TYPE=Release -S . -B "build"
```
the configuration looked like this:
```plain
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
-- The CXX compiler identification is MSVC 19.27.29111.0
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.28.0.windows.1")
-- git Version: v1.5.0-56f9d826
-- Version: 1.5.0
-- Performing Test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX -- success
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- failed to compile
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK -- success
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - not found
-- Found Threads: TRUE
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
-- Looking for Google Test sources
-- Looking for Google Test sources in C:/Users/Adam Badura/Dropbox/Projekty/adambadura-benchmark/googletest
CMake Error at CMakeLists.txt:34 (message):
  Did not find Google Test sources! Either pass correct path in
  GOOGLETEST_PATH, or enable BENCHMARK_DOWNLOAD_DEPENDENCIES, or disable
  BENCHMARK_ENABLE_GTEST_TESTS / BUILD_TESTING.


CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/ExternalProject.cmake:1772 (get_property):
  get_property could not find TARGET googletest.  Perhaps it has not yet been
  created.
Call Stack (most recent call first):
  CMakeLists.txt:54 (ExternalProject_Get_Property)


CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/ExternalProject.cmake:1774 (message):
  External project "googletest" has no SOURCE_DIR
Call Stack (most recent call first):
  CMakeLists.txt:54 (ExternalProject_Get_Property)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Adam Badura/Dropbox/Projekty/adambadura-benchmark/build/third_party/googletest/CMakeFiles/CMakeOutput.log".
CMake Error at cmake/GoogleTest.cmake:13 (message):
  CMake step for googletest failed: 1
Call Stack (most recent call first):
  CMakeLists.txt:276 (include)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Adam Badura/Dropbox/Projekty/adambadura-benchmark/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/Adam Badura/Dropbox/Projekty/adambadura-benchmark/build/CMakeFiles/CMakeError.log".
```

So, perhaps the instruction should include parameters set in a way, which doesn't result in an error. I will check this on Linux as well.

Reply:
I have made a test on Linux, including CMake 3.13.3 and CMake 3.7.1 (the closest to 3.5 I had readily available).

The commands were the following (after `README.md` and the command here):
```bash
git clone https://github.com/google/benchmark.git
git clone https://github.com/google/googletest.git benchmark/googletest
cd benchmark
cmake -E make_directory "build"
cmake -E chdir "build" cmake -DCMAKE_BUILD_TYPE=Release ../
cmake --build "build" --config Release
cmake -E chdir "build" ctest --build-config Release
```
and it did work just fine!

Reply:
As to my above comment:

> So, perhaps the instruction should include parameters set in a way, which doesn't result in an error. I will check this on Linux as well.

it turns out I was wrong. I do get this one, but only if I'm not doing the
```bash
git clone https://github.com/google/googletest.git benchmark/googletest
```
step, which is given in the `README.md`. I'm sorry!

Reply:
As to `--parallel <jobs>` I have both good news and bad news.

**The bad news** is that checking with CMake 3.7.1 made me realize that `cmake` in build mode supports the `--parallel` argument only starting from 3.12. So, our minimum required CMake 3.5 doesn't support it. Hence, we should remove it from the steps in `README.md`.

We can mention it in the `#` comment or free text after the commands since it seems likely users will have CMake >= 3.12.

We can also mention, that prior to CMake 3.12 adding `-- -j [$(nproc)]` to the build command will perform the build in parallel (making it _much_ faster!). However, those are build system specific commands and the `-j` works with Makefiles and Ninja, but not others.

**The good news** is `--parallel` does work on `ctest` even in CMake 3.5. (Well, I checked 3.7.1, however, it is documented on 3.5 as well.) However, unlike with `cmake` in build mode, there is no default for `<jobs>`, so you have to pass something (like `$(nproc)`) or else it will have no effect.

While adding `--parallel $(nproc)` to the test command speed it up 3 times in my case.

Sadly, the `$(nproc)` is system-specific, so again I would only mention it in the text.

Reply:
added the build config and the notes about parallel (short version pointing to the cmake docs).

thanks for the comprehensive comments!